### ISSUE TITLE
Return child Property of Object

### ISSUE THREAD

**Issue by whitecobalt** [ChilliCream/hotchocolate#3297]
TL:DR

I want the GraphQL server to return a child property from an object as the object. I.e. return the value in "foo.bar" as if it was "foo".

--------------

I am creating the GraphQL types and schema using Reflection. Works great.

I have one type in my code that encapsulates a Lazy Loaded object. Also works great. 

The Lazy Loader is a generic class, that exposes the object through a "Value" property. When "get" on "Value" is hit by the code, the loader checks to see if the object is loaded already, and if not loads and returns it.

For backwards compatibility, I **don't** want to alter the GraphQL schema so the front-end has to include "Value" to access the property. 

Otherwise this:
```
query
{
        Foo
        {
        totalCount
        nodes {
            ID
            Name
            ProjectID
            Project
            {
                ID
                Name
            }
        }
      }
  }
```

Would have to become:
```
query
{
        Foo
        {
        totalCount
        nodes {
            ID
            Name
            ProjectID
            Project
            {
                Value
                {
                     ID
                     Name
                }
            }
        }
      }
  }
```

To prevent this I am trying to expose the "Value" property as "Project" with the following code:
`
foreach (PropertyInfo pi in typeof(TEntity).GetProperties())
{
      descriptor.Field(pi.Name).Type(pi.PropertyType.GetGenericArguments()[0]).ResolveWith(pi.PropertyType.GetProperty("Value"));
}`

This compiles and runs, however the value returned is always null. I think this part `pi.PropertyType.GetProperty("Value")` is wrong, and causing the code to create a new instance of the object each time (rather than returning the instance already there).

I'm sure it's something really simple, but I can't figure out what!

Any help would be greatly appreciated.
